dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -27,9 +30,18 @@ def update
     existing_school_info = current_user.last_complete_school_info
     existing_school_info&.assign_attributes new_school_info_params
     if existing_school_info.nil? || existing_school_info.changed?
-      submitted_school_info = SchoolInfo.where(new_school_info_params).
-        first_or_create(validation_type: SchoolInfo::VALIDATION_NONE)
-      current_user.update! school_info: submitted_school_info
+      submitted_school_info =
+        if new_school_info_params[:school_id]
+          SchoolInfo.where(new_school_info_params).
+          first_or_create
+        else
+          SchoolInfo.where(new_school_info_params).
+          first_or_create(validation_type: SchoolInfo::VALIDATION_COMPLETE)
+        end
```

Comment:
It might be nice to have a comment here documenting why we only pass `VALIDATION_COMPLETE` when we don't have a `school_id`.